src/lifx/light.js

```diff
@@ -529,4 +529,162 @@ Light.prototype.colorZones = function(startIndex, endIndex, hue, saturation, bri
   this.client.send(packetObj, callback);
 };
 
+/**
+ * Requests tile getDeviceChain 701
+ * @param {Function} callback a function to accept the data
+ */
+Light.prototype.getDeviceChain = function(callback) {
+  validate.callback(callback, 'light getDeviceChain method');
+
+  const packetObj = packet.create('getDeviceChain', {}, this.client.source);
+  packetObj.target = this.id;
+  const sqnNumber = this.client.send(packetObj);
+  this.client.addMessageHandler('stateDeviceChain', function(err, msg) {
+    if (err) {
+      return callback(err, null);
+    }
+    return callback(null, msg);
+  }, sqnNumber);
+};
+
+/**
+ * Sets Tile Position
+ * @example light.setUserPosition(0, 12, 13, 0, () => {})
+ * @param {Number} tileIndex	unsigned 8-bit integer
+ * @param {Number} userX	32-bit float
+ * @param {Number} userY	32-bit float
+ * @param {Number} reserved	16-bit integer
+ * @param {Function} callback called when light did receive message
+ */
+Light.prototype.setUserPosition = function(tileIndex, userX, userY, reserved, callback) {
+  if (tileIndex === undefined || typeof tileIndex !== 'number') {
+    throw new TypeError('LIFX light setUserPosition tileIndex to be a number');
+  }
+  if (userX === undefined || typeof userX !== 'number') {
+    throw new TypeError('LIFX light setUserPosition userX to be a number');
+  }
+  if (userY === undefined || typeof userY !== 'number') {
+    throw new TypeError('LIFX light setUserPosition userY to be a number');
+  }
```

Comment:
These validations are now repeated on multiple functions and unnecessarily clutter the function body. Move these to new validation functions in `validator.js` similarly to what's been done for light zone support for example and then replace these with them.

Reply:
done